core/src/mindustry/input/InputHandler.java

```diff
@@ -488,8 +491,13 @@ public static void tileConfig(@Nullable Player player, Building build, @Nullable
 
             throw new ValidateException(player, "Player cannot configure a tile.");
         }
+
+        var previous = build.config();
         build.configured(player == null || player.dead() ? null : player.unit(), value);
-        Core.app.post(() -> Events.fire(new ConfigEvent(build, player, value)));
+
+        if(build.config() != previous){
```

Comment:
Don't call config() twice again, this will cause logic blocks to compress all their code again. Just use `value`.

Reply:
Kind of strange that the building is always configured and rate limit is always consumed but the event is only fired when the config has actually changed

Reply:
The problem is that `value` isn't always identical to `config()` result